File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    
    # 첫번째 값 리스트에 저장
    answer.append(arr[0])
    for num in arr: 
        if answer[-1] != num: # 리스트의 마지막 요소와 다를경우 저장
            answer.append(num)
    
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    int now = -1;
    for(int i = 0; i < arr.size(); i++){
        if (now != arr[i]){
            answer.push_back(arr[i]);
            now = arr[i];
        }
    }

    return answer;
}